src/storage/buffer_manager.cpp

```diff
@@ -810,7 +863,6 @@ void BufferManager::WriteTemporaryBuffer(block_id_t block_id, FileBuffer &buffer
 	}
 	// get the path to write to
 	auto path = GetTemporaryPath(block_id);
-	D_ASSERT(buffer.size > Storage::BLOCK_SIZE);
```

Comment:
Should we remove this assertion? While it is not required for correctness in general we want to avoid allocating small buffers that can be evicted, as writing tiny buffers to disk and buffer managing tiny buffers comes with significant overhead for little gain.